### ISSUE TITLE
Selectors with ";" are badly parsed or causes the parser to stuck

### ISSUE THREAD

**Issue by eramdam** [postcss/postcss-selector-parser#89]
The parser hangs up when trying to parse those two selectors:

``` css
#\;
.\;
```

These two are valids selectors so I guess they should be parsed correctly. Or at least throw an error.

Happy to give details if needed

**Comment by ben-eb** [comment]
Hi @eramdam,

I've tried this and these selectors don't fail the tests (or hang the parser), see this commit:

https://github.com/postcss/postcss-selector-parser/commit/4104b414af929be4cc3f8f2557306741e645819d

Do you have something that can exhibit the problem?

**Comment by eramdam** [comment]
Hey @ben-eb!

I can reproduce the issue with this snippet

```js
const selectorParser = require('postcss-selector-parser');
const getParsedSelector = (selectors, transform) => selectorParser(transform).process(selectors).result;

getParsedSelector('.\;', (selectors) => {
  console.log(selectors);
});
```

**Comment by ben-eb** [comment]
@eramdam Ah, I see. In JavaScript the backslash is the escape character, so if you want a literal backslash you always have to double that up (like I did with the tests). But I think that we should throw an error when an unescaped semicolon is encountered, as it is a syntax error. How does this sound?

> Expected a backslash preceding the semicolon.

Thanks!

**Comment by eramdam** [comment]
@ben-eb well that's much better to me than a stuck process so I'm all in :)

**Comment by ben-eb** [comment]
Released in `2.2.2`.
